cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java

```diff
@@ -126,6 +127,63 @@ public void testDeployUsingArtifact() throws Exception {
       doDelete(getVersionedAPIPath("apps/" + appId.getId(), appId.getNamespaceId())).getStatusLine().getStatusCode());
   }
 
+  @Test
+  public void testDeployVersionedAndNonVersionedApp() throws Exception {
+    Id.Artifact artifactId = Id.Artifact.from(Id.Namespace.DEFAULT, "configapp", "1.0.0");
+    addAppArtifact(artifactId, ConfigTestApp.class);
+
+    ApplicationId appId = new ApplicationId(Id.Namespace.DEFAULT.getId(), "cfgAppWithVersion", "version_1");
```

Comment:
We should allow versions such as `1.0.0`, etc. I think I saw that we enforce versionId to be an entity id above. Is that check required for version id @gokulavasan ?